agent/lib/kontena/workers/volumes/volume_manager.rb

```diff
@@ -102,9 +102,9 @@ def volume_exist?(volume_name)
 
     def terminate_volumes(current_ids)
       Docker::Volume.all.each do |volume|
```

Comment:
Bonus points for putting a `fail if current_ids.any? {|id| id.nil?}` here... my level of paranoia just says that it's too easy to end up with `nil`s somewhere by accident, and in this kind of place they would cause a lot of damage.